apps/src/code-studio/assets/SpriteUpload.js

```diff
@@ -1,31 +1,48 @@
 import React from 'react';
 import color from '@cdo/apps/util/color';
 
+const spriteLocation = {
+  LIBRARY: 'library',
+  LEVEL: 'level'
+};
+
 export default class SpriteUpload extends React.Component {
   state = {
     fileData: null,
     filePreviewURL: '',
     filename: '',
+    spriteAvailability: '',
     category: '',
+    currentCategories: [],
     uploadStatus: {
       success: null,
       message: ''
     }
   };
 
+  componentDidMount() {
+    fetch(`/api/v1/animation-library/manifest/spritelab/en_us`)
+      .then(response => response.json())
+      .then(data =>
+        this.setState({currentCategories: Object.keys(data.categories)})
+      );
+  }
+
   handleSubmit = event => {
     event.preventDefault();
 
-    return fetch(
-      `/api/v1/animation-library/level_animations/${this.state.filename}`,
-      {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'image/png'
-        },
-        body: this.state.fileData
-      }
-    )
+    let destination =
+      this.state.spriteAvailability === spriteLocation.LEVEL
+        ? `/level_animations/${this.state.filename}`
+        : `/spritelab/category_${this.state.category}/${this.state.filename}`;
```

Comment:
nit: destructuring properties from state will clean this ternary up a lot.

however, do we also want to make this a switch statement? if we added another value to the spriteLocation enum, would this have the behavior we want/expect?

Reply:
(yes to destructuring - I need to make that a habit!)

In regards to the switch, I was just thinking that I wasn't sure that there was a need for curriculum folks to add sprites to a third location . . . and then I remembered gamelab. lol. So yes, I think switching to a switch here future-proofs against when we might want to expand this to include gamelab sprites.